modApiServer/test/org/aion/api/server/ApiAionTests.java

```diff
@@ -0,0 +1,398 @@
+            List l1 = new ArrayList<ITransaction>();
+            l1.add(tx);
+            l1.add(tx);
+            l1.add(tx);
+            pendingRcvd.setFuncArgs(Collections.singletonList(l1));
+
+            ees.add(pendingRcvd);
+
+            EventTx pendingUpdate = new EventTx(EventTx.CALLBACK.PENDINGTXUPDATE0);
+            List l2 = new ArrayList<>();
+            l2.add(new AionTxReceipt());
+            l2.add(-1);
+            pendingUpdate.setFuncArgs(l2);
+
+            ees.add(pendingUpdate);
+
+            EventBlock evBlock = new EventBlock(EventBlock.CALLBACK.ONBLOCK0);
+            AionBlockSummary abs = new AionBlockSummary(null, null, null, null);
+            evBlock.setFuncArgs(Collections.singletonList(abs));
+
+            ees.add(evBlock);
+
+            //provokes exception in EpApi.run()
+            ees.add(new EventBlock(EventBlock.CALLBACK.ONBLOCK0));
+            ees.add(new EventDummy());
+        }
+
+    }
+
+    @Test
+    public void TestCreate() {
+        System.out.println("run TestCreate.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        System.out.println("API Version = " + api.getApiVersion());
+        assertNotNull(api.getInstalledFltrs());
+    }
+
+    @Test
+    public void TestInitNrgOracle() {
+        System.out.println("run TestInitNrgOracle.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        api.initNrgOracle(impl);
+        assertNotNull(api.getNrgOracle());
+        // Initing a second time should not create a new NrgOracle
+        api.initNrgOracle(impl);
+        assertNotNull(api.getNrgOracle());
+    }
+
+    @Test
+    public void TestStartES() {
+        System.out.println("run TestStartES.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        api.startES("thName");
+        api.addEvents();
+        try
+        {
+            Thread.sleep(5000);
+        }
+        catch(InterruptedException ex) {}
+        api.shutDownES();
+        assertTrue(api.allFlagsSet());
+    }
+
+    @Test
+    public void TestGetBlock() {
+        System.out.println("run TestGetBlock.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        assertNotNull(api.getBlockTemplate());
+
+        AionBlock blk = impl.getBlockchain().getBestBlock();
+
+        assertEquals(blk, api.getBestBlock());
+
+        assertEquals(blk, api.getBlock(-1));
+
+        assertEquals(blk.toString(), api.getBlockByHash(blk.getHash()).toString());
+        assertEquals(blk.toString(), api.getBlock(blk.getNumber()).toString());
+
+        Map.Entry rslt = api.getBlockWithTotalDifficulty(blk.getNumber());
+        assertEquals(rslt.getKey().toString(), blk.toString());
+        if (!blk.isGenesis())
```

Comment:
How come this if block is needed?  Shouldn't you be able to know whether blk.isGenesis() is expected to be true or false?

Reply:
i don't understand when the bestBlock is the genesis block...but it doesn't seem necessary, so i've gotten rid of the if block

Reply:
Sounds good